### PR TITLE
Handle pymongo.OperationFailure errors when updating indexes

### LINKED ISSUE #1568: Error updating genes on hasta stage
Maybe it's not a bug and stage is different from prod but I have a problem with indexes here.

The Ensembl service are working again and in order to fix #1557 I'm trying to update genes on hasta stage.

It seems it's working but then I got this error:
```
2019-12-03 11:26:59 hasta.scilifelab.se scout.load.transcript[203580] DEBUG Gene ENSG00000224567 does not exist in build 38
2019-12-03 11:26:59 hasta.scilifelab.se scout.load.transcript[203580] DEBUG Gene ENSG00000271375 does not exist in build 38
2019-12-03 11:26:59 hasta.scilifelab.se scout.load.transcript[203580] DEBUG Gene ENSG00000286009 does not exist in build 38
2019-12-03 11:26:59 hasta.scilifelab.se scout.load.transcript[203580] DEBUG Gene ENSG00000286009 does not exist in build 38
2019-12-03 11:26:59 hasta.scilifelab.se scout.load.transcript[203580] DEBUG Gene ENSG00000286009 does not exist in build 38
2019-12-03 11:26:59 hasta.scilifelab.se scout.load.transcript[203580] DEBUG Gene ENSG00000286009 does not exist in build 38
2019-12-03 11:26:59 hasta.scilifelab.se scout.load.transcript[203580] DEBUG Gene ENSG00000223517 does not exist in build 38
2019-12-03 11:26:59 hasta.scilifelab.se scout.load.transcript[203580] DEBUG Gene ENSG00000224518 does not exist in build 38
2019-12-03 11:26:59 hasta.scilifelab.se scout.load.transcript[203580] DEBUG Gene ENSG00000286201 does not exist in build 38
2019-12-03 11:26:59 hasta.scilifelab.se scout.load.transcript[203580] DEBUG Gene ENSG00000283076 does not exist in build 38
Building transcripts  [####################################]  100%             
2019-12-03 11:27:02 hasta.scilifelab.se scout.load.transcript[203580] INFO Loading transcripts...
2019-12-03 11:27:02 hasta.scilifelab.se scout.adapter.mongo.transcript[203580] INFO Loading transcript bulk
2019-12-03 11:27:14 hasta.scilifelab.se scout.load.transcript[203580] INFO Number of transcripts in build 38: 192946
2019-12-03 11:27:14 hasta.scilifelab.se scout.load.transcript[203580] INFO Number of transcripts with refseq identifier: 119959
2019-12-03 11:27:14 hasta.scilifelab.se scout.load.transcript[203580] INFO Number of primary transcripts: 24675
2019-12-03 11:27:15 hasta.scilifelab.se scout.adapter.mongo.index[203580] INFO Updating indexes...
2019-12-03 11:27:15 hasta.scilifelab.se scout.adapter.mongo.index[203580] INFO Adding index : build_chromosome
2019-12-03 11:27:17 hasta.scilifelab.se scout.adapter.mongo.index[203580] INFO Adding index : build_hgncid
2019-12-03 11:27:17 hasta.scilifelab.se scout.adapter.mongo.index[203580] INFO Adding index : build_aliases
2019-12-03 11:27:17 hasta.scilifelab.se scout.adapter.mongo.index[203580] INFO Adding index : caseid_variantrank
Traceback (most recent call last):
  File "/home/proj/bin/conda/envs/S_main/bin/scout", line 11, in <module>
    sys.exit(cli())
  File "/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/flask/cli.py", line 557, in main
    return super(FlaskGroup, self).main(*args, **kwargs)
  File "/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/flask/cli.py", line 412, in decorator
    return __ctx.invoke(f, *args, **kwargs)
  File "/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/scout/commands/update/genes.py", line 104, in genes
    adapter.update_indexes()
  File "/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/scout/adapter/mongo/index.py", line 70, in update_indexes
    self.db[collection_name].create_indexes(indexes)
  File "/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/pymongo/collection.py", line 1624, in create_indexes
    session=session)
  File "/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/pymongo/collection.py", line 245, in _command
    retryable_write=retryable_write)
  File "/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/pymongo/pool.py", line 517, in command
    collation=collation)
  File "/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/pymongo/network.py", line 125, in command
    parse_write_concern_error=parse_write_concern_error)
  File "/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/pymongo/helpers.py", line 145, in _check_command_response
    raise OperationFailure(msg % errmsg, code, response)
pymongo.errors.OperationFailure: Index with pattern: { hgnc_symbols: 1, rank_score: -1, category: 1, variant_type: 1 } already exists with different options
```

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR should fix #1568 and eventually #1557 

**How to test**:
1. Install the branch on Hasta stage and use the stage environment with `us`
1. Run the command:
`scout update genes`

**Expected outcome**:
The command should update genes and run until the end, with no errors but just warnings
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR

### CODE REVIEW

**Comment by northwestwitch** [comment]
Tested on Hasta and it works:

image

**Comment by northwestwitch** [comment]
Unfortunately when I tested this yesterday I didn't think to run the 
`scout update genes` within a screen session, so today when I checked I was disconnected from hasta stage and I don't know if it would have finished.

Now I runned again the command from within a screen session but now OMIM has temporarily blocked out key (This data account exceeded its download cap, please contact us at https://omim.org/contact if this is an issue). This is super annoying!

Anyway I'm quite sure that the fix works..

**Comment by northwestwitch** [comment]
Haha I'm running the command with @moonso 's key now and it works again! :)

**Comment by dnil** [comment]
This will make you rich eventually @moonso!

**Comment by northwestwitch** [comment]
The command works, an index-related warning is triggered:

image

**Comment by northwestwitch** [comment]
Any volunteer to review this one?

**Comment by dnil** [comment]
Sry I missed that comment! Will have a final look from the train if I can catch a seat later.

File: scout/adapter/mongo/index.py [hunk 1]
```diff
@@ -67,14 +69,19 @@ def update_indexes(self):
                 if index_name not in existing_indexes:
                     nr_updated += 1
                     LOG.info("Adding index : %s" % index_name)
-                    self.db[collection_name].create_indexes(indexes)
+                    try:
+                        self.db[collection_name].create_indexes([index])
+                    except pymongo_errors.OperationFailure as op_failure:
```
Comment (dnil) [inline scout/adapter/mongo/index.py]: :+1:

**Comment by dnil** [comment]
Does update diseases work now as well? Then we could get another issue closed: #1557

**Comment by northwestwitch** [comment]
Oh no I forgot to try the updat disease command! Almost on the train as well, will test from home this weekend

**Comment by dnil** [comment]
Also I think we should do the next update to prod - with disease and genes - early on a weekend as well just in case...

**Comment by northwestwitch** [comment]
Update diseases command was successful:

image

**Comment by northwestwitch** [comment]
Merging now!

**Pull request merged**